src/platform/mbed/PlatformManagerImpl.cpp

```diff
@@ -0,0 +1,287 @@
+#include <new>
+
+#include "platform/internal/CHIPDeviceLayerInternal.h"
+
+#include <inet/InetLayer.h>
+#include <platform/CHIPDeviceLayer.h>
+#include <platform/PlatformManager.h>
+#include <platform/ScopedLock.h>
+#include <platform/internal/GenericPlatformManagerImpl.cpp>
+#include <rtos/ThisThread.h>
+
+#include "MbedEventTimeout.h"
+
+#if CHIP_SYSTEM_CONFIG_USE_LWIP
+#include <lwip/tcpip.h>
+#endif
+
+#define DEFAULT_MIN_SLEEP_PERIOD (60 * 60 * 24 * 30) // Month [sec]
+
+using namespace ::chip;
+using namespace ::chip::Inet;
+using namespace ::chip::System;
+
+namespace chip {
+namespace DeviceLayer {
+
+// TODO: Event and timer processing is not efficient from a memory perspective.
+// Both occupy at least 24 bytes when only 4 bytes is required.
+// An optimized designed could use a separate circular buffer to store events
+// and register a single mbed event in the event queue to process all of them.
+// A similar design can be used for timers.
```

Comment:
Is it not advantageous though to integrate directly into the platform event queue?

If this is refactored as the comment suggests, you'll break serialization if any code (callbacks presumably) enqueue events using native interfaces.

Reply:
From a memory standpoint, depending on the events it might be less efficient to use the _native_ mbed event queue compared to the approach proposed in the note. Mbed event queue store function like objects which are at least 24 bytes large. Chip events are usually smaller. 

If the note is implemented then events can be executed in a different order than queued if the event queue is used by other components or directly rather than with `PlatformMgr().PostEvent`. That's ok as the event queue used here is local to this platform layer: we can make sure events are queued PostEvent and not with the _native_ call.  

We've been happy with current memory usage and aren't actively planning on implementing the note referenced above. But I think it is a good thing to keep it in if work in this area is required latter.